pkg/agent/controller/networkpolicy/packetin.go

```diff
@@ -278,17 +271,62 @@ func (c *Controller) rejectRequest(pktIn *ofctrl.PacketIn) error {
 	binary.BigEndian.PutUint32(icmpData[:ICMPUnusedHdrLen], 0)
 	copy(icmpData[ICMPUnusedHdrLen:], ipHdr[:ipHdrLen+8])
 	return c.ofClient.SendICMPPacketOut(
-		srcMAC.String(),
-		dstMAC.String(),
+		srcMAC,
+		dstMAC,
 		srcIP,
 		dstIP,
 		inPort,
-		-1,
+		outPort,
 		isIPv6,
 		icmpType,
 		icmpCode,
 		icmpData,
-		true)
+		packetOutType)
+}
+
+// isServiceTraffic uses ServiceEPStateField to decide if this is a Service
+// traffic, when AntreaProxy is enabled. And when AntreaProxy is disabled, if the
+// destination IP of the reject response packet is on the local Node, but the
+// source MAC is antrea-gw's MAC, then this is a rejection of Service traffic.
+// Because this response is heading to kube-proxy.
+func (c *Controller) isServiceTraffic(pktIn *ofctrl.PacketIn, antreaProxyEnabled bool, dstMAC string, isDstLocal bool) bool {
+	if antreaProxyEnabled {
+		matches := pktIn.GetMatches()
+		match := getMatchRegField(matches, openflow.ServiceEPStateField)
+		if match != nil {
+			svcEpstate, err := getInfoInReg(match, openflow.ServiceEPStateField.GetRange().ToNXRange())
+			return err == nil && svcEpstate&openflow.EpSelectedRegMark.GetValue() == openflow.EpSelectedRegMark.GetValue()
```

Comment:
is the `err != nil` case meant to be fallthrough like this?

Reply:
If err != nil it should return false. Changed to a clearer way.